presto-main/src/main/java/com/facebook/presto/sql/planner/SubfieldUtils.java

```diff
@@ -53,6 +54,16 @@ public static SubfieldPath deferenceOrSubscriptExpressionToPath(Node expression)
             else if (expression instanceof SubscriptExpression) {
                 SubscriptExpression subscript = (SubscriptExpression) expression;
                 Expression index = subscript.getIndex();
+                if (index instanceof Cast) {
+                    // map['str'] can be represented as a
+                    // SubscriptExpression with an index of cast of a
+                    // string literal to a varchar(xx).
+                    Cast cast = (Cast) index;
```

Comment:
The following comment might have been lost:

---
Can this block be simplified to index = ((Cast) index).getExpression();?